packages/flutter/lib/src/widgets/value_updater.dart

```diff
@@ -0,0 +1,171 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'framework.dart';
+
+/// A builder function used by the [ValueUpdater] widget that takes the current
+/// value and returns a widget.
+typedef ValueUpdaterBuilder<T> = Widget Function(BuildContext context, T value);
+
+/// A widget that updates a value for the widgets built using a builder.
+///
+/// The [ValueUpdater] widget can be used to avoid the need for creating a
+/// stateful widget just to trigger a change in another widget on creation.
+///
+/// If not given a [valueNotifier], will start at the [initialValue] (or the
+/// [initialValue], if given) and toggle to the [value] on the frame after the
+/// widget is first built. It will only toggle once.
+///
+/// If given a [valueNotifier], will start at [initialValue] and
+/// switch to the value of the [valueNotifier], and then continue to switch
+/// to each new value as the [valueNotifier] changes.
+///
+/// Like all stateful widgets in Flutter, if this widget is recreated, it will
+/// lose its current state. To prevent this, give it a [Key] to allow it to be
+/// reused instead of recreated.
+///
+/// {@tool dartpad}
+/// This example shows how to use the [ValueUpdater] widget to trigger an
+/// [AnimatedAlign] animation when a widget is shown for the first time. This is
+/// done without needing to create a stateful widget.
+///
+/// ** See code in examples/api/lib/widgets/value_updater/value_updater.0.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to use the [ValueUpdater] widget to trigger an
+/// [AnimatedAlign] animation when a widget is shown for the first time, and
+/// then change the value of the alignment and animate each time a button is
+/// pressed. This is done without needing to create a stateful widget.
+///
+/// ** See code in examples/api/lib/widgets/value_updater/value_updater.1.dart **
+/// {@end-tool}
+
+class ValueUpdater<T> extends StatefulWidget {
+  /// Creates a [ValueUpdater] widget that updates the value based on
+  /// [valueNotifier] changes.
+  ///
+  /// The [builder], [initialValue], and [valueNotifier] are
+  /// required.
```

Comment:
nit: this doesn't match the signature below where valueNotifier seems to be optional?